modules/api/php/endpoints/candidate/visit/instrument/flags.class.inc

```diff
@@ -127,6 +126,104 @@ class Flags extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PUT request
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request) : ResponseInterface
+    {
+        // TODO :: Check permissions. How??
+        $this->_instrumentStatus = new \NDB_BVL_InstrumentStatus();
+        $this->_instrumentStatus->select($this->_instrument->commentID);
+
+        if (!$this->_instrument->determineDataEntryAllowed()) {
+            return new \LORIS\Http\Response\JSON\Forbidden(
+                'Can not update instruments that are flagged as complete.'
+            );
+        }
+
+        $data = json_decode((string) $request->getBody(), true);
+
```

Comment:
In the case of a PUT request, the shape of the request data should contain all the fields. It needs to be a complete representation of the resource.

_The HTTP PUT request method creates a new resource or replaces a representation of the target resource with the request payload._

as opposed to

_The HTTP PATCH request method applies partial modifications to a resource._

Reply:
The data requested now need to contain all the fields, otherwise an error that tells which entries are missing is displayed.